XiEditor/StyleMap.swift

```diff
@@ -100,7 +100,8 @@ struct StyleSpan {
 }
 
 func utf8_offset_to_utf16(_ s: String, _ ix: Int) -> Int {
-    return s.utf8.index(s.utf8.startIndex, offsetBy: ix).samePosition(in: s.utf16)!._offset
+    //TODO: (swift 4 conversion: verify that this works as expected)
```

Comment:
@raphlinus This is the one bit I'm not totally sure about.

Reply:
Looks good to me, and I tested the utf8/utf16 conversion with emoji. I was thrown off slightly by the fact that gestures were broken, but that was unrelated to this change (google/xi-editor#404).